File: 3. DSA/02. C++/01. Basics/Question08.cpp
// Sum of square of first 'n' natural numbers

#include <iostream>
using namespace std;

int main()
{
    int num;
    cout << "Enter a number: ";
    cin >> num;

    int sum = 0;
    for(int i = 1; i <= num; i++)
    {
        sum += i * i;
    }

    // int sum = num * (num + 1) / 2 * (2 * num + 1) / 3;

    cout << "\nThe sum of square of first " << num << " natural numbers is " << sum << endl;
}

File: 3. DSA/04. Python/01. Basics/Question08.py
# Sum of square of first 'n' natural numbers

n = int(input("Enter the number: "))

#* Method 1
sum = n * (n + 1) // 2 * (2 * n + 1) // 3
print("Sum of square of first 'n' natural numbers:", sum)

#* Method 2
sum = 0
for i in range(n + 1):
    # sum += i * i;
    sum += i ** 2;

print("Sum of square of first 'n' natural numbers:", sum)